regolith/builders/beamplanbuilder.py

```diff
@@ -0,0 +1,98 @@
+"""Builder for Current and Pending Reports."""
+from regolith.builders.basebuilder import LatexBuilderBase
+from regolith.tools import all_docs_from_collection
+from regolith.tools import group
+import pandas as pd
+
+
+class BeamPlanBuilder(LatexBuilderBase):
+    """
+    Build a file of experiment plans for the beamtime from database entries. The report is in .tex file. The template
+    of the file is in the 'templates/beamplan.txt'. The data will be grouped according to beamtime. Each beamtime
+    will generate a file of the plans. If 'beamtime' in 'rc' are not None, only plans for those beamtime will be
+    generated.
+    """
+    btype = "beamplan"
+    needed_dbs = ['beamplan', "people"]
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        gtx["beamplan"] = all_docs_from_collection(rc.client, "beamplan")
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+
+    @staticmethod
+    def gather_info(docs):
+        """
+        Make a table as the summary of the plans and a list of experiment plans. The table header contains: serial
+        id, person name, number of sample, sample container, sample holder, measurement, estimated time (min). The
+        latex string of the table will be returned. The plans contain objective, steps in preparation,
+        steps in shipment, steps in experiment and a to do list. The latex string of the paragraphs will be returned.
+
+        Parameters
+        ----------
+        docs : list
+            A list of documents.
+
+        Returns
+        -------
+        info : dict
+        The information obtained from the database and formatted. It contains the key value pairs:
+            table The latex string of table.
+            plans The list of experiment plans. Each experiment plan is a list of strings.
+
```

Comment:
I don't think an empty line is needed here, but please check the numpy format

Reply:
Fixed/